src/main/java/de/tum/in/www1/artemis/service/LegalDocumentService.java

```diff
@@ -0,0 +1,178 @@
+     * @return the updated privacy statement
+     */
+    public PrivacyStatement updatePrivacyStatement(PrivacyStatement privacyStatement) {
+        return (PrivacyStatement) updateLegalDocument(privacyStatement);
+    }
+
+    /**
+     * Returns the legal document if you want to update it.
+     * If it currently doesn't exist an empty string is returned as it will be created once the user saves it.
+     *
+     * @param language the language of the legal document
+     * @param type     the type of the legal document
+     * @return the legal document with the given language and type
+     */
+    private LegalDocument getLegalDocumentForUpdate(LegalDocumentLanguage language, LegalDocumentType type) {
+        if (getLegalDocumentPathIfExists(language, type).isEmpty()) {
+            return switch (type) {
+                case PRIVACY_STATEMENT -> new PrivacyStatement("", language);
+                case IMPRINT -> new Imprint("", language);
+            };
+
+        }
+        return readLegalDocument(language, type);
+
+    }
+
+    /**
+     * Returns the legal document if you want to view it
+     * If it currently doesn't exist in the given language, the other language is returned.
+     * If it currently doesn't exist in any language, an exception is thrown.
+     *
+     * @param language the language of the legal document
+     * @param type     the type of the legal document
+     * @return the legal document with the given language and type
+     */
+    private LegalDocument getLegalDocument(LegalDocumentLanguage language, LegalDocumentType type) {
+        // if it doesn't exist for one language, try to return the other language, and only throw an exception if it doesn't exist for both languages
+        if (getLegalDocumentPathIfExists(LegalDocumentLanguage.GERMAN, type).isEmpty() && getLegalDocumentPathIfExists(LegalDocumentLanguage.ENGLISH, type).isEmpty()) {
+            throw new BadRequestAlertException("Could not find " + type + " file for any language", "legalDocument", "noLegalDocumentFile");
+        }
+        else if (language == LegalDocumentLanguage.GERMAN && getLegalDocumentPathIfExists(language, type).isEmpty()) {
+            language = LegalDocumentLanguage.ENGLISH;
+        }
+        else if (language == LegalDocumentLanguage.ENGLISH && getLegalDocumentPathIfExists(language, type).isEmpty()) {
+            language = LegalDocumentLanguage.GERMAN;
+        }
+
+        return readLegalDocument(language, type);
+
+    }
+
+    private LegalDocument readLegalDocument(LegalDocumentLanguage language, LegalDocumentType type) {
+        String legalDocumentText;
+        try {
+            legalDocumentText = Files.readString(getLegalDocumentPath(language, type));
+        }
+        catch (IOException e) {
+            log.error("Could not read {} file for language {}", type, language);
+            throw new InternalServerErrorException("Could not read " + type + " file for language " + language);
```

Comment:
Teamscale

Reply:
Technically it would be better to pass the exception to the constructor of the `RuntimeException` in the end. However, a lot of change would be needed for this, which I believe is not worth it here